crates/nargo_cli/src/cli/prove_cmd.rs

```diff
@@ -100,16 +101,18 @@ pub(crate) fn prove_with_path<B: Backend, P: AsRef<Path>>(
                 &program.bytecode,
             )
             .map_err(CliError::CommonReferenceStringError)?;
-            (common_reference_string, program)
+            (common_reference_string, program, None, None)
         }
         None => {
-            let program = compile_circuit(backend, program_dir.as_ref(), compile_options)?;
+            let (program, context) =
+                compile_circuit(backend, program_dir.as_ref(), compile_options)?;
+            let debug = program.debug.clone();
```

Comment:
Can we avoid cloning here? Perhaps `preprocess_program` needs to return the DebugInfo as well?

Reply:
I've changed it to return DebugInfo -- we'll most likely need to rethink this interface, though I'd rather do it after we have completely separated compilation from optimizations